### PR TITLE
Move to recent version of Gradle  (7.6.1) and latest Dataflow Beam SDK (2.42.0)

### PR DESCRIPTION
**Pull request by annadata**
Move to gradle recent version (7.6.1) and Dataflow Beam SDK to latest (2.42.0)

Testing done:
1. Gradle compile works with latest gradle version 8.0.1
2. Ran the CSV(on GCS) -> BQ pipeline on Dataflow.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -15,17 +15,15 @@
  */
 buildscript {
     ext {
-        dataflowBeamVersion = '2.33.0'
+        dataflowBeamVersion = '2.42.0'
```
Comment (slilichenko) [inline build.gradle:18]: Latest Bam version is 2.46.0 - https://beam.apache.org/get-started/downloads/

File: build.gradle [hunk 2]
```diff
@@ -83,30 +80,30 @@ repositories {
...
-    compile group: 'org.apache.beam', name: 'beam-sdks-java-io-amazon-web-services', version: dataflowBeamVersion
-    compile "com.google.auto.value:auto-value-annotations:1.6.2"
-    compile 'com.google.cloud:google-cloud-storage:2.1.9'
+    implementation group: 'org.apache.beam', name: 'beam-runners-direct-java', version: dataflowBeamVersion
+    implementation group: 'org.apache.beam', name: 'beam-sdks-java-extensions-ml', version: dataflowBeamVersion
+    implementation group: 'org.apache.beam', name: 'beam-sdks-java-io-amazon-web-services', version: dataflowBeamVersion
+    implementation group: 'org.slf4j', name: 'slf4j-jdk14', version: '1.7.5'
+    implementation 'com.google.cloud:google-cloud-kms:0.70.0-beta'
+    implementation 'com.google.guava:guava:27.0-jre'
+    implementation group: 'com.google.cloud', name: 'google-cloud-dlp', version: '3.0.11'
```
Comment (slilichenko) [inline build.gradle:94]: Latest - 3.16.0 (https://mvnrepository.com/artifact/com.google.cloud/google-cloud-dlp)

File: build.gradle [hunk 3]
```diff
@@ -83,30 +80,30 @@ repositories {
...
+    implementation 'com.google.api-client:google-api-client:1.27.0'
```
Comment (slilichenko) [inline build.gradle:95]: Latest - 2.2.0 (https://mvnrepository.com/artifact/com.google.api-client/google-api-client)

**Review by slilichenko — COMMENTED** [review]
Looks like there are newer versions of several libraries than what is upgraded in this PR. I haven't checked all of them; just did a couple.

File: build.gradle (hunk 1 above)
Comment (annadata) [reply build.gradle]: Done.

File: build.gradle (hunk 3 above)
Comment (annadata) [reply build.gradle]: Done

**Review by slilichenko — APPROVED** [review]
LGTM

**Pull request merged**